src/rabbitalerter.js

```diff
@@ -0,0 +1,208 @@
+/**
+ * An alerting system for message queues.
+ * @module rabbitalerter
+ */
+const assert = require('assert');
+const taskcluster = require('taskcluster-client');
+
+/**
+ * An interface used to generate various TaskCluster Notifications.
+ *
+ * @class RabbitAlerter
+ */
+class RabbitAlerter {
+  /**
+   * @param {number} config.alerter.messageCountTolerance       - When a queue holds more messages
+   *                                                              than this tolerance, an alert will be created.
+   * @param {number} config.alerter.messagePublishRateTolerance - When the number of messages published per second
```

Comment:
Thanks for the clarification @jonasfj!

The message below is just to verify my understanding.
Let me know if I am missing anything, or if what I say is inaccurate.

In a nutshell, when the publishing rate of messages to a queue is higher than that queue's consumption rate, then we send an alert. (If the publish rate is higher than the consumption rate, eventually, the system will fail due to too many messages in the queue). 

We also do not want to send too many false positive alerts. For example, a queue has zero consumption rate, and one message is being published to it a day. However, since the publish rate is higher than the consumption rate, alerts would be sent either way.

I think a solution for this would be to set a threshold tolerance for alerting where we take the following:
**message publish rate / message consumption rate**. 

Once this tolerance is exceeded, then we can send an alert. Of course, there are exceptions where there are intervals where the message consumption rate will be zero. 

When the consumption rate is zero and the publish rate is greater than zero, when we wait 8 minutes. If no messages have been consumed, then this queue is deleted.